presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry_provider.py

```diff
@@ -46,24 +50,182 @@ def __init__(
+
+        The method adds the NLP recognizer to the list of recognizers
+        if it is not already present,
+        or removes it if it is not enabled in the configuration.
+        Furthermore, it checks if there are
+        any inconsistencies in configuration. For example:
+        - Multiple enabled NLP recognizers in the configuration for one language.
+        - The NLP recognizer in the configuration does not match the Nlp Engine.
+
+        :param recognizers: List of recognizers to update.
+        :param recognizers_conf: Configuration of the recognizers from the YAML file
+        :param supported_languages: List of supported languages.
+
+        :raises ValueError: If there are multiple enabled NLP recognizers
+        in the configuration.
+        :raises ValueError: If the NLP recognizer
+        in the configuration does not match the Nlp Engine.
+        """
+        nlp_engine = self.nlp_engine
+
+        if not nlp_engine:
+            return
+
+        for language in nlp_engine.get_supported_languages():
+            self.__update_based_on_nlp_recognizer_conf_and_lang(
+                recognizers=recognizers, nlp_engine=nlp_engine, language=language
+            )
+            self.__remove_disabled_nlp_recognizers(
+                recognizers=recognizers,
+                recognizers_conf=recognizers_conf,
+                language=language,
+            )
+
+    @staticmethod
+    def __update_based_on_nlp_recognizer_conf_and_lang(
+        recognizers: List[EntityRecognizer],
+        nlp_engine: NlpEngine,
+        language: str,
+    ):
+        """
+        Update the list of recognizers with nlp recognizers.
+
+        Update based on the NLP recognizer configuration for a specific language.
+        """
+
+        nlp_recognizers = [
+            rec
+            for rec in recognizers
+            if isinstance(rec, SpacyRecognizer) and rec.supported_language == language
+        ]
+
+        # Case 1: NLP recognizer is not in the list of recognizers
+        if not nlp_recognizers:
+            warning_text = (
+                f"NLP recognizer (e.g. SpacyRecognizer, StanzaRecognizer) "
+                f"is not in the list of recognizers "
+                f"for language {language}. "
+                f"Adding the default {SpacyRecognizer.__name__} to the list."
+                f"If you wish to remove the NLP recognizer, "
+                f"define it as `enabled=false`."
+            )
+            logger.warning(warning_text)
+            warnings.warn(warning_text)
+            if nlp_engine:
+                recognizers.append(
+                    SpacyRecognizer(
+                        supported_language=language,
+                        supported_entities=nlp_engine.get_supported_entities(),
+                    )
+                )
+            else:
+                recognizers.append(SpacyRecognizer(supported_language=language))
```

Comment:
Add a test for when there is no nlp_engine

Reply:
Added